### PR TITLE
Making the component update call more tolerant, allowing for image name in argument

### PR DESCRIPTION
**Pull request by marians**
Fix for Issue https://github.com/giantswarm/cli/issues/378

If a user does a component update giving the last argument to `swarm update <component> <image>:<tag>` (instead of only `<tag>`), we parse the argument and check if the image name prepended is identical with the current one. If yes, we proceed with the update, else quit with an error message.

Example for non-matching image:

```
$ swarm update flighttracking/redis mysql:3.1
ERROR: Bad image name 'mysql' provided.
The component uses the image 'redis' and can only be updated to versions (tags) of this image.
```

Depends on:
- https://github.com/giantswarm/generic-types-go/pull/8

### CODE REVIEW

**Comment by marians** [comment]
RFR @ewoutp @zeisss

File: update.go [hunk 1]
```diff
@@ -123,6 +131,48 @@ func updateCmdRun(cmd *cobra.Command, args []string) {
 	}
 }
 
+// Takes strings in the format "<tag>" or "<image>:<tag>"
```
Comment (zeisss) [inline update.go]: please start the docstring with the name of the function

File: update.go [hunk 2]
```diff
@@ -123,6 +131,48 @@ func updateCmdRun(cmd *cobra.Command, args []string) {
...
+		}
+		definition, err := userconfig.ParseServiceDefinition([]byte(strDefinition))
+		if err != nil {
+			return "", err
+		}
+		if definition.Components[userconfig.ComponentName(componentName)].Image == nil {
+			// The component doesn't even have an image definition.
+			return "", errors.New(updateComponentNoImageMsgTemplate)
+		}
+		currentImage := definition.Components[userconfig.ComponentName(componentName)].Image.String()
```
Comment (zeisss) [inline update.go]: How about this 

```
currentImageTag := definition.Components[userconfig.ComponentName(componentName)].Image.Version
```

No need to parse, combine and split again.

Comment (marians) [inline update.go]: But I'm interested in the part _before_ the `: `, not the version/tag.

Comment (zeisss) [inline update.go]: Then I would still favor we add a function for this in the `DockerImage` type. @ewoutp has a similar requirement for the `imageverifyd` and is manually combining fields.

Comment (marians) [inline update.go]: Any idea for the naming? DockerImage.EveryhtingButTheVersionPart()?

Comment (marians) [inline update.go]: I settled for `DockerImage.UnversionedString()` as a companion for `DockerImage.String()`. See https://github.com/giantswarm/generic-types-go/pull/8

**Comment by marians** [comment]
Now includes the suggested change to `generic-types-go` and the required cleanups.

**Comment by zeisss** [comment]
LGTM

**Comment by marians** [comment]
So how does merging work now?

`generic-types-go` was in version 1.3.2. Do I tag it using `builder` and then point the CLI's Makefile to the new version (1.3.3 probably)?

**Comment by zeisss** [comment]
Exactly. Merge the dependency, use builder to make a release, put it in the Makefile here, merge here.

**Pull request merged**